src/ensembl/src/shared/badged-button/BadgedButton.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import defaultStyles from './BadgedButton.scss';
+import classNames from 'classnames';
+
+type Props = {
+  children: React.ReactChild;
+  badge?: string | number | undefined;
```

Comment:
If the component explicitly says in its name that it is a BadgedButton, then why should the badge field be optional (or its value be undefined)?

Reply:
It is optional because a value for the badge might not always be available. Let's take unread messages count for example, it will have a value set only when there are some unread messages.

Reply:
Fair point.

There are two other alternatives:

a) Choose in the parent component which button to show — either the one with a badge, or the one without, based on some conditions
b) Use the `showBadge: boolean` prop.

Alternative (a) is probably too cumbersome; but alternative (b) might actually be better than the current implementation. Especially given that the jsx:

`{props.badge && <div className={className}>{props.badge}</div>}`

will show a "0" (in a wrong place, too) if there are 0 unread messages, while you will probably not want to show any badge at all.

Reply:
> will show a "0" (in a wrong place, too) if there are 0 unread messages

I think `props.badge` would return false if the badge value is 0 and not render the badge.

Do we still require an additional parameter when the parent can actually control wether to show the badge or not, just by setting the right value?

Reply:
> I think props.badge would return false if the badge value is 0 and not render the badge.

Try it 😄 
if `props.badge` is 0, it will return 0; and since 0 is falsy, it will not evaluate the second part of the expression and render 0. At least that's what it _should_ do.

Reply:
Ahh! yes, that's right. I thought of it like a standard if condition. 
I could alternatively change it to:
`{props.badge? <div className={className}>{props.badge}</div> : null}`

Reply:
What if you want to display the 0? Is this something we need to consider?

Reply:
> What if you want to display the 0? Is this something we need to consider?

That's why we allow the type string. We can use '0' to display 0 and also ' ' to display an empty badge.

> Otherwise, just casting 0 to boolean (e.g. Boolean(props.badge), or the slightly less explicit !!props.badge) will work

Yes, this seems like a better solution.

Reply:
> That's why we allow the type string. We can use '0' to display 0 and also ' ' to display an empty badge.

Erm. I don't know if we want to be able to display a `0` in the badge (and if we don't, then we don't care); but just in case we do, what will the code look like?

```
<BadgedButton
  badgeContent={props.messageCount}  
>
 <Button>Something</Button>
</BadgedButton>
```

If `props.messageCount` is a number, then 0 will be a number (and will not show up). So if we want it to show up, will we to cast `props.messageCount` to string like this?

```
<BadgedButton
  badgeContent={`${props.messageCount}`}  
>
 <Button>Something</Button>
</BadgedButton>
```

Reply:
I don't think there will be many cases where we'll be displaying a 0. If we need to do in some cases, I guess it's better to leave it to the parent component to pass it as a string instead. 
So yes, we will have to cast it to a string.